hyperspy/models/edsmodel.py

```diff
@@ -0,0 +1,881 @@
+            The Xray lines. If 'all_alpha', fit all using all alpha lines
+        **kwargs : extra key word arguments
+            All extra key word arguments are passed to fit or
+            multifit, depending on the value of kind.
+
+        """
+
+        if calibrate == 'resolution':
+            free = self._twin_xray_lines_width
+            fix = self.fix_xray_lines_width
+            scale = self._set_energy_resolution
+        elif calibrate == 'scale':
+            free = self._twin_xray_lines_scale
+            fix = self.fix_xray_lines_energy
+            scale = self._set_energy_scale
+        elif calibrate == 'offset':
+            free = self._twin_xray_lines_offset
+            fix = self.fix_xray_lines_energy
+            scale = self._set_energy_offset
+        ref = free(xray_lines=xray_lines)
+        self.fit(**kwargs)
+        fix(xray_lines=xray_lines)
+        scale(xray_lines=xray_lines, ref=ref)
+        self.update_plot()
+
+    def free_sub_xray_lines_weight(self, xray_lines='all', bound=0.01):
+        """
+        Free the weight of a sub X-ray lines
+
+        Remove the twin on the height of sub-Xray lines (non alpha)
+
+        Parameters
+        ----------
+        xray_lines: list of str or 'all'
+            The Xray lines. If 'all', fit all lines
+        bounds: float
+            Bound the height of the peak to a fraction of
+            its height
+        """
+        def free_twin():
+            component.A.twin = None
+            component.A.free = True
+            if component.A.value - bound * component.A.value <= 0:
+                component.A.bmin = 1e-10
+                # print 'negative twin!'
+            else:
+                component.A.bmin = component.A.value - \
+                    bound * component.A.value
+            component.A.bmax = component.A.value + \
+                bound * component.A.value
+            component.A.ext_force_positive = True
+        xray_families = [
+            utils_eds._get_xray_lines_family(line) for line in xray_lines]
+        for component in self:
+            if component.isbackground is False:
+                if xray_lines == 'all':
+                    free_twin()
+                elif utils_eds._get_xray_lines_family(
+                        component.name) in xray_families:
+                    free_twin()
+
+    def fix_sub_xray_lines_weight(self, xray_lines='all'):
+        """
+        Fix the weight of a sub X-ray lines to the main X-ray lines
+
+        Establish the twin on the height of sub-Xray lines (non alpha)
+        """
+        def fix_twin():
+            component.A.bmin = 0.0
+            component.A.bmax = None
+            element, line = utils_eds._get_element_and_line(component.name)
+            for li in elements_db[element]['Atomic_properties']['Xray_lines']:
+                if line[0] in li and line != li:
+                    xray_sub = element + '_' + li
+                    component_sub = self[xray_sub]
```

Comment:
I asked @francisco-dlp about this (because I wasn't 100% sure what to put in the try) he suggested instead using a check of whether the component is in the model with an if statement so that when it breaks you really know what's gone wrong where a try function may cause errors that are more tricky to trace - so that's what I've implemented.